### DISCUSSION TITLE
beforeSuite and afterSuite being called only once and not per describe in a spec file

### DISCUSSION THREAD

**Discussion by rsinghi** [webdriverio/webdriverio#11156] [Q&A]
My understanding is that the afterSuite function in WebDriverIO is called after each describe block execution. Same for beforeSuite.
However, I am noticing that for the following usecase where I have multiple describes in a spec, the afterSuite and beforeSuite are being called only one for DESC1. (though all the tests are being executed):
```
describe('DESC 1: ', function() {
    beforeEach(async function() {
    });

    it('Test 1', async () => {
        console.log('In Test 1');
    });

});

describe('DESC 2: ', function() {
    beforeEach(async function() {
    });

    it('Test 2', async () => {
        console.log('In TEST 2 ');
    });
});
```

I am printing the following in afterSuite:
    afterSuite: function (suite) {
        console.log(`Finished running: ${suite.title}`);
    },

The output is: Finished running: DESC 1
There is no output for DESC 2.

**Comment by christian-bromann** [comment]
This is a known issue, I recommend only to have one single root `describe`, e.g.:

```ts
describe('my test', () => {
    describe('DESC 1: ', function() {
        beforeEach(async function() {
        });
    
        it('Test 1', async () => {
            console.log('In Test 1');
        });
    
    });
    
    describe('DESC 2: ', function() {
        beforeEach(async function() {
        });
    
        it('Test 2', async () => {
            console.log('In TEST 2 ');
        });
    });
});
```
